### PR TITLE
[docs] Added details on how to use the engflowapis

### PR DESCRIPTION
**Pull request by anfelbar**
This is the begging on how to use the repository. Nonetheless, we will add some sections explaining what does each of the
APIs do. In particular a section for

- [auth](https://github.com/EngFlow/engflowapis/tree/main/engflow/auth): What does this do?
- [eventstore/v1](https://github.com/EngFlow/engflowapis/tree/main/engflow/eventstore/v1): What does this do?
- [notification/v1](https://github.com/EngFlow/engflowapis/tree/main/engflow/notification/v1): What does this do?

@Yannic those sections need your contribution.
I believe we need to document in such a way that any customer with any language can use the apis without going to any other repository.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,21 +1,91 @@
...
+```
+
+This command will fetch a fresh copy of the repository and link it with
+the name `com_engflow_engflowapis`.
+
+### Check out a full example
+
+You should use the protocol buffer definitions from the `engflowapis`. In here,
+you need to include building blocks such as `protocol buffer` tools and `googleapis` definitions.
+Check out the full working [engflow example](https://github.com/EngFlow/example),
```
Comment (jayconrod) [inline README.md:90]: Link to https://github.com/EngFlow/example/tree/main/java/com/engflow/notificationqueue maybe?

File: README.md [hunk 2]
```diff
@@ -1,21 +1,91 @@
...
-> Note: older versions of Bazel may work as well, but are not officially supported.
+## Using the APIs in your (bazel) project
 
-First, [install bazel](https://docs.bazel.build/versions/master/install.html).
+In order to integrate the `engflowapis` into your Bazel project, you should
+include it using one of two bazel methods; 
+`git_repository` or `http_archive`.
 
-To build all libraries:
+### Download with git_repository
```
[the remark this answers is not in the document]
Comment (anfelbar) [reply README.md:31]: I would like not to force people to take one or the other option. Bazel provides the manner so I guess we document it. Maybe we add something like "we strongly recommend, whenever possible, the use of http_archive".

File: README.md [hunk 3]
```diff
@@ -1,21 +1,72 @@
...
 
-To build all libraries:
+In order to integrate the `engflowapis` into your Bazel project, you should
+include it using `http_archive`.
+
+We recommend to download a specific commit from GitHub and then store and reference
+it in any storage service. To do so, you may download a repository `zip` file using a
+commit sha:
 
 ```
```
Comment (petemounce) [inline README.md:37]: Please add syntax hint to every opening code fence?

**Comment by petemounce** [comment]
I'm good. Leaving to @jayconrod to do final pass because timezone closer.

File: README.md (hunk 3 above)
Comment (anfelbar) [reply README.md]: Done. Bazel is not yet supported by the markdown language, right?

File: README.md [hunk 4]
```diff
@@ -1,21 +1,91 @@
...
+```
+
+This command will fetch a fresh copy of the repository and link it with 
+the name `com_engflow_engflowapis`. Be aware that this command downloads all
+history from the repository, but you can configure it to fetch only 
+a given commit (see [git_repository](https://bazel.build/rules/lib/repo/git) for more info).
+
+### Download with http_archive
+
+Alternatively, you may use a specific commit from GitHub and then store and reference
```
[the remark this answers is not in the document]
Comment (jayconrod) [reply README.md]: Let's try to make this even shorter and simpler. No need to give a `wget` command. No need to mention `local_repository` at all. Just give the `WORKSPACE` code to paste and maybe a short sentence introducing it.

> In order to integrate the `engflowapis` into your Bazel project, you can include it using `http_archive`. For example:
>
> ```
> ...workspace code...
> ```

File: README.md (hunk 3 above)
Comment (jayconrod) [reply README.md]: I've mostly seen `bzl` used as the syntax hint. Not sure if that's actually meaningful to GitHub's parser though.

File: README.md [hunk 5]
```diff
@@ -1,21 +1,72 @@
...
+This, for instance, downloads the repository in its commit sha `47aa858b498da13e7863356aaef9c6d05da0a7f2`.
+The repository may be used locally using Bazel's [local_repository](https://bazel.build/reference/be/workspace#local_repository).
+However, it is recommended to store the file in a highly available storage service in such a way
+that you can use `http_archive` like this
+
+
+```
+...
+load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive")
+...
```
Comment (jayconrod) [inline README.md:50]: Nit: remove the `...` dots to make this slightly easier to copy/paste.

File: README.md [hunk 6]
```diff
@@ -1,21 +1,72 @@
 # EngFlow APIs
 
-This repository contains the gRPC service definitions for public EngFlow APIs.
+This repository contains the gRPC service definitions for public EngFlow APIs:
+
+- **authentication API**
+- **notification queue API**
+- **eventstore API**.
```
Comment (jayconrod) [inline README.md:7]: Nit: remove trailing period on this line or add it to the two lines above.

**Review by jayconrod — APPROVED** [review]

**Review by Yannic — CHANGES_REQUESTED** [review]
Source-of-truth for this code is the main repo

**Comment by anfelbar** [comment]
> Source-of-truth for this code is the main repo

Okay, I will finish the changes asked by @jayconrod  and then @Yannic  can help me on adding the result to the main repo. I guess it is synced by copybara but I don't have knowledge on that yet.

**Comment by anfelbar** [comment]
Changed [PR](https://github.com/EngFlow/engflow/pull/9625) to engflow repo.

File: README.md (hunk 3 above)
Comment (petemounce) [reply README.md]: https://github.com/github/linguist/blob/master/lib/linguist/languages.yml#L6253-L6272

**Pull request closed without merging**